crmsh/ui_node.py

```diff
@@ -332,18 +332,24 @@ def do_standby(self, context, *args):
                 standby_template_str = standby_template.format(node_id=node_id, value="on")
                 xml_item.append(etree.fromstring(standby_template_str))
 
-        cib_str = xmlutil.xml_tostring(cib)
+        self._do_standby_or_online_action(cib, node_list, context.get_command_name())
+
+    def _do_standby_or_online_action(self, cib, node_list, action):
+        """
+        Common part of standby and online
+        """
         # Consider both "nodes" and "status" section might contain different standby value at the same time
-        # Should replace all possible "off" values here
         for node in node_list:
             node_id = utils.get_nodeid_from_name(node)
-            cib_str = re.sub(constants.STANDBY_NV_RE.format(node_id=node_id, value="off"), r'\1value="on"\2', cib_str)
+            for query_path in [constants.XML_NODE_QUERY_STANDBY_PATH, constants.XML_STATUS_QUERY_STANDBY_PATH]:
```

Comment:
According to the newly expected `lifetime`, to prevent potential interference,  we should probably delete any possibly existing setting with a different `lifetime`?

Reply:
Action `standby` only accept two kinds of lifetime: "reboot" or "forever", and these would make cib.xml no different with no lifetime set

Reply:
For sure without explicitly specifying `lifetime`, it goes into the configuration meaning it's `forever`.

What I meant:

If an user wants to have a "reboot" standby, while there's an existing `standby` setting in the configuration no matter it's on or off. They won't actually get the "reboot" standby because the code here basically sets standby in both configuration and status.

That's the reason why I suggest:

1) If setting standby in configuration, drop any from status.
2) if setting standby in status, drop any from configuration.

Reply:
@gao-yan  Thanks for pointing these!
Changed